### ISSUE TITLE
Feature request: Select Level

### ISSUE THREAD

**Issue by Richard-L** [rr-/Tomb1Main#197]
One of the PS1 favourites we began discussing in more detail today.

Curiously it's called "Select Level" instead of "Level Select", which we usually say. The game reads the save data to then know how many levels you've beaten, and will offer you to (re)start from all currently "unlocked" ones. Essentially you can correct your game's progress course if you find you've collected too little pick-ups, or missed out on a weapon at some point. It's also just nice to access any previous level of your liking. Finally it's nice to see your so far achievement in the game documented.

Perhaps some examples first. Would be fantastic if someone could post Saturn equivalents should there be any differences.

Here's a completed PS1 TR1 with all levels available:

https://user-images.githubusercontent.com/4934209/140825340-158a71f8-0510-4cd0-a58a-6a84185e577a.mp4

^ notice the little arrows appearing on the top or bottom to signal more rows. Depending on a user's text scaling setting, they could come in handy.

Compare also OpenLara's implementation of this feature, where everything is visible in one bigger window:

https://user-images.githubusercontent.com/4934209/140825733-5696fd58-e915-49f6-9130-214c3a90bd5b.mp4

(the "Current Position" feature we ignore in this issue)

Just for added reference, this is what "Select Level" looks like on PS1 if not all levels are unlocked:

https://user-images.githubusercontent.com/4934209/140825829-6212c977-0f86-4221-9bc5-ea5e43e65aed.mp4

**Some notes:**

- The inventory seems to be included per level, based on the player's progress of the given playthrough.
- If an FMV is part of the level, it is played first before the level begins:

https://user-images.githubusercontent.com/4934209/140826502-28235ab3-5202-40b3-8b77-36b3a71c3033.mp4

- If the game was beaten and NG+ is activated, selecting one of the past levels seems to force NG+ mode onto them. Ie weapons have unlimited ammo like pistols, but collected medipack counts are still intact. @Trxyebeep could you confirm this is correct?

https://user-images.githubusercontent.com/4934209/140827104-b20fe65d-399b-486a-9fbf-29c5ce62701f.mp4

As such, Select Level seems like a list of savestates the PS1 collects when you complete a level, and choose to save after the stats screen.

The PC way of saving kind of obliterates this system, as it mixes "Select Level" and "Current Position" into one fluent "omnipresent" saving. Thus perhaps should there be a JSON setting called "**Choose saving style: PC (=0), PS1 (=1)**"? I'm not quite sure. Do share some opinions.

Related issue is "Restart Level", which essentially just loads the save taken at the end of the previous level: https://github.com/rr-/Tomb1Main/issues/48

**Comment by walkawayy** [comment]
I think this is blocked by #277. We will have to keep track of every level's completion information for every savegame slot. This would allow level selection (and level restart) back to the selected level with the player's items/stats for that particular playthrough.

**Comment by Richard-L** [comment]
If the PS1/Saturn way of handling this (=autosave at level end of previous level) was implemented, and these saves were then only available via "select level", would that not work?

Is the added complication that you want to reinstate more than one playthroughs, depending on where a savestate stems from?

**Comment by walkawayy** [comment]
Yes. Theoretically, all the PC version saveslots could be different playthroughs. On the PS1, you have a save from each level start and a current position for only one playthrough.

**Comment by rr-** [comment]
Duplicate of #33. Closing the older ticket as this one contains more information.

**Comment by Richard-L** [comment]
@walkawayy I've been trying to make thoughts on your solution to the missing passport page dilemma. I understand you want to make the "save" page have two subpages, one for PC or console style loading respectively. Could you post a video of it in operation perhaps?

**Comment by walkawayy** [comment]
https://streamable.com/jlpv4h

**Comment by Richard-L** [comment]
Many thanks. This is slightly confusing at first, but at the same time it's super nice! The little arrows are totally crucial here in my opinion, as else the UX wouldn't indicate any connection between savegame and unlocked levels.

I had one thought, although it's a certain departure from the game's UI aesthetic: What about showing the two lists side by side, where the right one with the unlocked levels would just update based on what savestate you're hovering on on the left. I wonder if this would be an even stronger indicator of level select _depending_ on the savestate.

Another thought for later: I think it would be incredible if we conceived a concession of levels from the two groups of "Atlantean Scion" and "Unfinished Business". Is it necessary to think about potential ramifications here for something like this?
